### PR TITLE
v1.0.0a3 — Fix Tags Exclusion in the Router YAML Round-Trip (RFP-003)

### LINKED ISSUE #28: RFP-003 — Fix Tags Exclusion in the Router YAML Round-Trip
**Status:** Draft · **Domain:** `tiferet-openapi` ·
**Branch:** `v1.x-proto` · **Issue:** (unpublished)
**Related:** `docs/core-domain-distillation.md` §8
**Depends on:** none
**Blocks:** none

## Summary
`ApiRouteYamlObject`'s `to_data.yaml` role excludes `'tags'` (`tiferet_openapi/mappers/openapi.py:97` on `main`), so a route's `tags` never survive a YAML write. This is unrelated to spec generation — it is a mapper round-trip defect that happened to be fixed in the same out-of-process commit (`v1.x-proto` PR #22) as the `generate_spec` wiring (RFP-002). This RFP isolates and reconstructs that one-line fix on its own.

## Motivation
Bundling an unrelated correctness fix into a feature RFP (as PR #22 did) makes both harder to review independently and obscures which change is a defect fix versus a scope decision. `tags` is also a field `generate_spec` will read once RFP-002 lands (`operation['tags'] = route.tags` when present); if a route's tags never round-trip through a YAML write, that read finds nothing even when a route was declared with tags, silently. Fixing this now, independent of RFP-002, means RFP-002 does not have to carry a mapper fix inside a context-generation RFP.

## Current state
`ApiRouteYamlObject._ROLES['to_data.yaml']` (`tiferet_openapi/mappers/openapi.py:93-99` on `main`) is:
```yaml
'to_data.yaml': {
    'by_alias': True,
    'exclude': {'id', 'endpoint', 'tags'},
}
```
`id` and `endpoint` are correctly excluded — both are derived at map time (`endpoint` from `{router_name}.{route_id}`, `mappers/openapi.py:193-200`; `id` from the YAML mapping key itself) and would be redundant or wrong if written back literally. `tags`, `summary`, `description`, `request_model`, and `response_model` are declared fields with no such derivation — `tags` is the only one of the five excluded from the write role, and comparing against `summary`/`description`/`request_model`/`response_model` (none of which appear in the exclusion set) confirms this is an inconsistency, not a deliberate design choice.

`TransferObject.to_primitive` (`tiferet/mappers/core.py:120-143`) merges `_ROLES[role]` kwargs into `model_dump(**kwargs)`; an excluded field is dropped from the dumped dict entirely, so a route written to YAML via this role loses its `tags` value regardless of what was set on the in-memory model.

On `v1.x-proto` (PR #22), the same line reads `'exclude': {'id', 'endpoint'}` — `tags` was removed from the set, with no other change to the role.

## Proposal
1. Remove `'tags'` from `ApiRouteYamlObject._ROLES['to_data.yaml']['exclude']` (`mappers/openapi.py:97`), matching `v1.x-proto`'s existing fix exactly.
2. Add a round-trip test: construct an `ApiRouteYamlObject` (or `ApiRouteAggregate` mapped through `from_model`) with a non-empty `tags` list, serialize via `to_primitive(role='to_data.yaml')`, and assert `tags` is present with its original value. The existing test suite's absence of such a test is why this defect shipped unnoticed for two releases (`v0.1.1`/`v0.1.2`) before proto happened to fix it as a side effect of unrelated work.

## Out of scope
- Any other field's exclusion set — `id`/`endpoint` remain excluded for `ApiRouteYamlObject`; `name` remains excluded for `ApiRouterYamlObject` (`mappers/openapi.py:163-166`), and neither is in question here.
- `generate_spec`'s consumption of `tags` — that is RFP-002.

## Risks and open questions
1. **Resolved.** No behavioral risk: removing a field from an `exclude` set only adds it to what's written; no existing consumer can be relying on `tags` being silently dropped, since that would mean relying on data loss.

## Acceptance criteria
- `ApiRouteYamlObject._ROLES['to_data.yaml']['exclude']` is `{'id', 'endpoint'}`.
- A route constructed with a non-empty `tags` list round-trips that value through `to_primitive(role='to_data.yaml')`.
- No other `_ROLES` entry in `mappers/openapi.py` changes.

## Suggested TRD slicing
Single TRD — this is a one-line fix plus one test; no natural seam to slice further.

### PR DESCRIPTION
**Pull request by Ashatz**
## What changed
- Verified `ApiRouteYamlObject._ROLES['to_data.yaml']['exclude']` on `v1.x-proto` already equals `{'id', 'endpoint'}` (matches the AC; `tags` was already removed via PR #22, as noted in the issue's "Current state" section). No production code change was needed for this.
- Added dedicated test coverage in `tiferet_openapi/mappers/tests/test_openapi.py`:
  - `test_api_route_yaml_object_to_data_yaml_excludes_only_id_and_endpoint` — asserts the exclude set literally equals `{'id', 'endpoint'}`.
  - `test_api_route_tags_round_trip_to_data_yaml` — constructs a route with a non-empty `tags` list and asserts it survives `to_primitive(role='to_data.yaml')`.
- No other `_ROLES` entries in `mappers/openapi.py` were changed.

## Acceptance criteria
- [x] `ApiRouteYamlObject._ROLES['to_data.yaml']['exclude']` is `{'id', 'endpoint'}`.
- [x] A route constructed with a non-empty `tags` list round-trips that value through `to_primitive(role='to_data.yaml')`.
- [x] No other `_ROLES` entry in `mappers/openapi.py` changes.

## Test notes
`pytest tiferet_openapi/ -v` currently hits pre-existing collection errors in `contexts/tests/test_openapi.py` and `repos/tests/test_openapi.py` due to a tiferet dependency version mismatch — this is RFP-001's (#26) scope, reproduced identically on unmodified `v1.x-proto`, and untouched here. Excluding those two files, the full suite passes: 31 passed.

Closes #28

Co-Authored-By: Warp <agent@warp.dev>

### CODE REVIEW

File: tiferet_openapi/mappers/tests/test_openapi.py [hunk 1]
```diff
@@ -298,3 +298,38 @@ def test_api_route_yaml_object_map_without_swagger_fields() -> None:
     assert aggregate.tags == []
     assert aggregate.request_model is None
     assert aggregate.response_model is None
+
+
```
Comment (Ashatz) [inline tiferet_openapi/mappers/tests/test_openapi.py]: please replace \n\n with \n between test artifacts please.

Comment (Ashatz) [reply tiferet_openapi/mappers/tests/test_openapi.py]: Fixed in f0f5e2f — removed both double-blank-line spots between the new test artifacts (single empty line between artifacts per tiferet-code-style). Full suite still green (61 passed).

**Comment by Ashatz** [comment]
Confirmed the double-blank-line spacing fix (commit f0f5e2f) — single empty line between test artifacts now, matching tiferet-code-style. Reran the full suite: 61 passed.
